### PR TITLE
Bump swift version to 5.10

### LINKED ISSUE #171: Bump swift version
### Feature Request

Bump `.swift-version` to latest version to allow usage of latest swift features

### PR DESCRIPTION
**Pull request by DelevoXDG**
## Stack

- #172
- #166
- #167
- #170
- #173
- #174
- #175
- #178
- #182
- #180 
- #181

## Describe your changes

- Bump `swift-tools-version` to `5.10`
- Bump version used in `.swift-version` to `5.10.0`
- Bump runner: use `macos-14` instead of `macos-12`
- Format using 5.10 tools

## Linked issues

Closes #171 

## Breaking changes

- [x] This issue contains breaking changes

- XCode 15.3+ (macOS 14+) is now required to build the package

### CODE REVIEW

**Comment by DelevoXDG** [comment]
Blocked for now, let's ensure this is not an issue for SDK users

**Comment by DelevoXDG** [comment]
Additionally blocked by https://github.com/actions/runner-images/issues/9459

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -5,12 +5,15 @@ on:
...
   test_and_lint:
-    runs-on: macos-12
+    runs-on: macos-14
     env:
       DEVNET_SHA: fa1238e8039a53101b5d2d764d3622ff0403a527
     steps:
       - uses: actions/checkout@v3
-
+      - name: Setup Xcode 15.3
+        uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd # v1.6.0
```
Comment (ddoktorski) [inline .github/workflows/checks.yml:14]: nit: Is there any specific reason to use the exact commit?
```suggestion
        uses: maxim-lobanov/setup-xcode@v1.6.0
```

**Review by ddoktorski — APPROVED** [review]

File: .github/workflows/checks.yml (hunk 1 above)
Comment (DelevoXDG) [reply .github/workflows/checks.yml:14]: For safety reasons mostly, a tag can be hypothetically replaced, unlike a commit hash.

**Review by THenry14 — APPROVED** [review]

**Pull request merged**